gnd/src/main/java/com/google/android/gnd/ui/home/AddFeatureDialogFragment.java

```diff
@@ -21,107 +21,74 @@
 
 import android.app.Dialog;
 import android.os.Bundle;
+import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.appcompat.app.AlertDialog;
 import androidx.fragment.app.FragmentManager;
 import com.google.android.gnd.R;
-import com.google.android.gnd.model.AuditInfo;
 import com.google.android.gnd.model.Project;
-import com.google.android.gnd.model.feature.Feature;
-import com.google.android.gnd.model.feature.Point;
 import com.google.android.gnd.model.layer.Layer;
-import com.google.android.gnd.persistence.uuid.OfflineUuidGenerator;
-import com.google.android.gnd.rx.Loadable;
-import com.google.android.gnd.system.auth.AuthenticationManager;
 import com.google.android.gnd.ui.common.AbstractDialogFragment;
-import com.google.android.gnd.ui.home.mapcontainer.MapContainerViewModel;
-import com.google.android.gnd.ui.map.CameraPosition;
 import com.google.common.collect.ImmutableList;
 import dagger.hilt.android.AndroidEntryPoint;
-import io.reactivex.Maybe;
-import io.reactivex.subjects.MaybeSubject;
 import java8.util.Objects;
 import javax.inject.Inject;
+import timber.log.Timber;
 
 @AndroidEntryPoint
 public class AddFeatureDialogFragment extends AbstractDialogFragment {
+
   private static final String TAG = AddFeatureDialogFragment.class.getSimpleName();
-  private final OfflineUuidGenerator uuidGenerator;
-  private final AuthenticationManager authManager;
 
-  private final MaybeSubject<Feature> addFeatureRequestSubject = MaybeSubject.create();
-  private HomeScreenViewModel homeScreenViewModel;
-  private MapContainerViewModel mapContainerViewModel;
+  @Nullable private Project project;
+  @Nullable private LayerSelectedListener listener;
 
   @Inject
-  public AddFeatureDialogFragment(
-      OfflineUuidGenerator uuidGenerator, AuthenticationManager authManager) {
-    this.uuidGenerator = uuidGenerator;
-    this.authManager = authManager;
-  }
-
-  @Override
-  public void onCreate(@Nullable Bundle savedInstanceState) {
-    super.onCreate(savedInstanceState);
-    // TODO: Move into new AddFeatureDialogViewModel?
-    this.homeScreenViewModel = getViewModel(HomeScreenViewModel.class);
-    this.mapContainerViewModel = getViewModel(MapContainerViewModel.class);
-  }
+  public AddFeatureDialogFragment() {}
 
-  public Maybe<Feature> show(FragmentManager fragmentManager) {
+  public void show(
+      @NonNull Project project,
+      @NonNull FragmentManager fragmentManager,
+      @NonNull LayerSelectedListener listener) {
```

Comment:
Instead of defining a custom functional interface, you should be able to use `Consumer<Layer>`.